### ISSUE TITLE
1.6.2: git_odb_open symbol was removed

### ISSUE THREAD

**Issue by 0-wiz-0** [libgit2/libgit2#6519]
git_odb_open is missing in the 1.6.2 release. It was there in 1.5.1.

In 1.5.1:
```
# nm /usr/pkg/lib/libgit2.so | grep git_odb_open 
00000000000c6a0c T git_odb_open
00000000000c93dd T git_odb_open_rstream
00000000000c8e53 T git_odb_open_wstream
```
In 1.6.2:
```
# nm /usr/pkg/lib/libgit2.so | grep git_odb_open                                                                                          
00000000000cb40e T git_odb_open_rstream
00000000000cae77 T git_odb_open_wstream
```
It's still in the header file:
```
/usr/pkg/include/git2/odb.h:GIT_EXTERN(int) git_odb_open(
```
(though there will be a signature change when `GIT_EXPERIMENTAL_SHA256` is enabled, which is a different problem).

Since this was not mentioned in the release notes for 1.6.0, 1.6.1, nor 1.6.2 and the prototype is still in the header, I guess this is an oversight.

**Comment by ethomson** [comment]
Ugh. Yes, that's a bug. Thanks for reporting it.

> (though there will be a signature change when GIT_EXPERIMENTAL_SHA256 is enabled, which is a different problem).

What's the problem with this? That it's a different API is by design.

If you want experimental SHA256 support, the entire API has changed. That's why it's opt-in.

This will eventually be the new API for libgit2 v2.0 once it's complete and working. 😓

**Comment by 0-wiz-0** [comment]
Sorry, I'm just used to "once you publish a function prototype, it's fixed forever".
It sounds like libgit2 will need source code changes for all users of the API. Perhaps it should be libgit3 then ;) Anyway, not really relevant for this issue, I'll shut up now :)

**Comment by ethomson** [comment]
Yeah, it's unfortunate. We could do some terrible bisection of the APIs to try to avoid breakage. But I'd like to see people complaining before I commit to that. (This serves as a datapoint. 😄)

**Comment by ethomson** [comment]
To expand on that - my thinking is that most people are not using libgit2, they're using LibGit2Sharp, rugged, nodegit, etc. and one of those needing to change once is not too onerous.

It will be interesting to see the delta between the APIs when SHA256 is in a usable state. Maybe we can just slap an `_ext` at the end of all the function names and call it a day with backcompat.

**Comment by ethomson** [comment]
Apologies again for the oversight - I missed this during refactoring. Should be fixed by #6520. Since this is the second symbol we missed, I did diff the exported symbol names between 1.5 and 1.6 and this was the only other one. 😓

**Comment by 0-wiz-0** [comment]
Thanks for the fix, and for checking all symbols! Good idea :)
